pkg/url/url_test.go

```diff
@@ -1,29 +1,37 @@
-package git_test
+package url_test
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
-	"github.com/aquasecurity/vexhub-crawler/pkg/crawl/git"
+	"github.com/aquasecurity/vexhub-crawler/pkg/url"
 )
 
-func TestNormalizeURL(t *testing.T) {
+func TestURL_GetterString(t *testing.T) {
 	tests := []struct {
-		name    string
-		rawURL  string
-		want    string
-		wantErr string
+		name        string
+		rawURL      string
+		want        string
+		wantSubDirs string
+		wantErr     string
 	}{
 		{
 			name:   "happy path - GitHub URL",
 			rawURL: "https://github.com/user/repo",
 			want:   "git::https://github.com/user/repo.git?depth=1",
 		},
 		{
-			name:   "happy path - GitHub URL with tree",
-			rawURL: "https://github.com/user/repo/tree/main/subfolder",
-			want:   "git::https://github.com/user/repo.git//subfolder?depth=1&ref=main",
+			name:        "happy path - GitHub URL with tree",
+			rawURL:      "https://github.com/user/repo/tree/main/subfolder/subfolder2",
+			want:        "git::https://github.com/user/repo.git?depth=1&ref=main",
+			wantSubDirs: "subfolder/subfolder2",
+		},
+		{
+			name:        "happy path - GitHub URL with subdirs",
+			rawURL:      "https://github.com/hashicorp/go-getter.git//testdata",
```

Comment:
@knqyf263 I'm not sure about this - do we need to remove the `subdirs` suffix in this case?

Reply:
I don't think the subdir URL format is specified, but we can leave it.

Reply:
Fixed in https://github.com/aquasecurity/vexhub-crawler/pull/28/commits/9517722832daf36c2004b63602a68da9b4dd7763